### PR TITLE
call pook.off in the end of pook.activate wrapper function

### PR DESCRIPTION
**Pull request by shift0965**
This PL is for this [issue](https://github.com/h2non/pook/issues/144)

### CODE REVIEW

**Comment by sarayourfriend** [comment]
Thank you for the issue and PR @shift0965 :pray: Can you add a unit test for this fix? You will also need to update `activate_async` to fix the same problem for async tests: https://github.com/h2non/pook/blob/master/src/pook/activate_async.py#L26

We'll need a unit test for each as well. If you would like guidance on adding the tests, let me know and I can give some pointers.

**Review by sarayourfriend — CHANGES_REQUESTED** [review]
Thank you for the issue and PR @shift0965 :pray: Can you add a unit test for this fix? You will also need to update `activate_async` to fix the same problem for async tests: https://github.com/h2non/pook/blob/master/src/pook/activate_async.py#L26

We'll need a unit test for each as well. If you would like guidance on adding the tests, let me know and I can give some pointers.

**Comment by shift0965** [comment]
Thank you for you feedback! I just added test cases for each scenario and let me know if there are any further adjustments you like me to make.

File: tests/unit/api_test.py [hunk 1]
```diff
@@ -47,3 +48,30 @@ def test_mock_contructors(engine):
...
+        api.get("foo.com")
+        assert engine.active is True
+        assert engine.isdone() is False
+
+    test_activate()
+    assert engine.active is False
+    assert engine.isdone() is True
+
+
+def test_activate_as_decorator_for_async(engine):
```
Comment (sarayourfriend) [inline tests/unit/api_test.py:66]:
```suggestion
async def test_activate_as_decorator_for_async(engine):
```

**Review by sarayourfriend — APPROVED** [review]
LGTM! One small change needed to get the tests to run (the test function needs to be async), but I'll commit that for you and then CI should pass and I can merge and release today :slightly_smiling_face: 

Thanks again for the contribution :tada:

File: tests/unit/api_test.py [hunk 2]
```diff
@@ -47,3 +48,30 @@ def test_mock_contructors(engine):
 
     assert len(engine.mocks) == 0
     assert engine.active is False
+
+
+def test_activate_as_decorator(engine):
+
```
Comment (sarayourfriend) [inline tests/unit/api_test.py:54]:
```suggestion
```

File: tests/unit/api_test.py [hunk 3]
```diff
@@ -47,3 +48,30 @@ def test_mock_contructors(engine):
...
+        assert engine.active is True
+        assert engine.isdone() is False
+
+    test_activate()
+    assert engine.active is False
+    assert engine.isdone() is True
+
+
+async def test_activate_as_decorator_for_async(engine):
+
```
Comment (sarayourfriend) [inline tests/unit/api_test.py:67]:
```suggestion
```

**Pull request merged**